### PR TITLE
Able to filter Cloud objects with glob notation.

### LINKED ISSUE #5772: Support regex filter to filter input objects for StaticS3Firehose
It would be nice if the regex filter is supported for StaticS3Firehose like for [LocalFirehose](http://druid.io/docs/latest/ingestion/firehose.html#localfirehose).

### PR DESCRIPTION
**Pull request by didip**
Fixes #5772.

### Description

In a heterogeneous environment, sometimes you don't have control over the input folder. Upstream can put any folder they want. In this situation the `S3InputSource.java` is unusable.

Most people like me solved it by using Airflow to fetch the full list of parquet files and pass it over to Druid. But doing this explodes the JSON spec. We had a situation where 1 of the JSON spec is 16MB and that's simply too much for Overlord.

This patch allows users to pass `{"filter": "*.parquet"}` and let Druid performs the filtering of the input files.

I am using the `glob` notation to be consistent with the [LocalFirehose syntax](https://druid.apache.org/docs/latest/ingestion/native-batch.html#localfirehose).

<hr>

##### Key changed/added classes in this PR
 * `S3InputSource`
 * `CloudObjectInputSource`
 * `OssInputSource`
 * `AzureInputSource`
 * `GoogleCloudStorageInputSource`

<hr>

This PR has:
- [x] been self-reviewed.
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [x] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

File: core/src/main/java/org/apache/druid/data/input/impl/CloudObjectInputSource.java
Comment (abhishekagarwal87) [inline core/src/main/java/org/apache/druid/data/input/impl/CloudObjectInputSource.java:51]: since many extensions extend this class, can you leave the old constructor too?

**Review by abhishekagarwal87 — COMMENTED** [review]
Thank you for this PR @didip. I don't see docs changes here. can you please add those too? I also don't see any tests for the new functionality. can you add some please?

**Comment by didip** [comment]
@abhishekagarwal87 Thank you for the review, I will work on those.

**Comment by didip** [comment]
@abhishekagarwal87 tests are added.

File: core/src/main/java/org/apache/druid/data/input/impl/CloudObjectInputSource.java
Comment (abhishekagarwal87) [inline core/src/main/java/org/apache/druid/data/input/impl/CloudObjectInputSource.java:72]: can this moved to a common private method method?

**Review by abhishekagarwal87 — COMMENTED** [review]
Thank you.  I have one minor comment. can you also look into CI failures? They seem to be related. PR looks good to me otherwise.

File: core/src/main/java/org/apache/druid/data/input/impl/CloudObjectInputSource.java
Comment (didip) [reply core/src/main/java/org/apache/druid/data/input/impl/CloudObjectInputSource.java]: @abhishekagarwal87 done. I will check the CI.

**Comment by didip** [comment]
@abhishekagarwal87 Specifically, which checks that you want me to take a look? The maven tasks don't show the actual reports. I don't know how to set these up yet.

**Comment by didip** [comment]
LGTM passed locally for me.
```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  06:51 min
[INFO] Finished at: 2022-06-16T22:11:30-07:00
[INFO] ------------------------------------------------------------------------
```

**Comment by didip** [comment]
@abhishekagarwal87 the checks passed.

**Comment by abhishekagarwal87** [comment]
@didip - travis check didn't run at all for your PR. I am going to close and reopen your PR to get that check running.

**Comment by didip** [comment]
@abhishekagarwal87 sorry for this dumb question, but where is the checkstyle-result file? I ran this: `${MVN} checkstyle:checkstyle --fail-at-end` locally, got two errors but I cannot find where is the actual report.

**Comment by didip** [comment]
`${MVN} spotbugs:check --fail-at-end -pl '!benchmarks'` failed due to unrelated `druid-sql` error.

**Comment by abhishekagarwal87** [comment]
@didip - I usually see the error on the terminal itself. maybe try removing `fail-at-end` parameter and you can look at the tail output.

**Comment by didip** [comment]
@abhishekagarwal87 ah, I finally found it with `-X` and grep for `[ERROR]`

**Comment by abhishekagarwal87** [comment]
cool. I am waiting for travis woes to be fixed and will merge your PR after that.

**Review by abhishekagarwal87 — APPROVED** [review]

**Comment by didip** [comment]
@abhishekagarwal87 is everything good now?

**Comment by abhishekagarwal87** [comment]
@didip - please check if travis failures are genuine.

**Comment by didip** [comment]
checking...

**Comment by didip** [comment]
@abhishekagarwal87 how can I get the actual report of `MAVEN_OPTS='-Xmx3000m' ${MVN} ${MAVEN_SKIP} dependency:analyze -DoutputXML=true -DignoreNonCompile=true -DfailOnWarning=true`?

I get a bunch of debug messages but nothing tells me on which file the dependency error happened.

**Comment by didip** [comment]
When I ran `MAVEN_OPTS='-Xmx3000m' ${MVN} ${MAVEN_SKIP} dependency:analyze`, it is completed.
```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  03:11 min
[INFO] Finished at: 2022-06-18T14:29:00-07:00
[INFO] ------------------------------------------------------------------------
```

**Comment by didip** [comment]
@abhishekagarwal87 the code coverage is not detecting my tests on `illegalArgsChecker`. Not sure what to do next from here.

**Comment by didip** [comment]
@abhishekagarwal87 I have all green code coverage, but why Travis still broken?

**Comment by didip** [comment]
73E57B45-A36F-49F0-954E-44AF88CAB33E

**Comment by didip** [comment]
@abhishekagarwal87 This test: https://app.travis-ci.com/github/apache/druid/jobs/574017537 have nothing to do with the files I touched.

```bash
	at org.apache.druid.testing.utils.ITRetryUtil.retryUntil(ITRetryUtil.java:94) ~[druid-integration-tests-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	at org.apache.druid.tests.indexer.AbstractStreamIndexingTest.doTestIndexDataWithAutoscaler(AbstractStreamIndexingTest.java:360) ~[test-classes/:?]
	at org.apache.druid.tests.parallelized.ITKafkaIndexingServiceNonTransactionalParallelizedTest.testKafkaIndexDataWithWithAutoscaler(ITKafkaIndexingServiceNonTransactionalParallelizedTest.java:62) ~[test-classes/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_292]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_292]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_292]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_292]
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132) ~[testng-7.3.0.jar:?]
	at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:599) [testng-7.3.0.jar:?]
	at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:174) [testng-7.3.0.jar:?]
	at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46) [testng-7.3.0.jar:?]
	at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:822) [testng-7.3.0.jar:?]
	at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:147) [testng-7.3.0.jar:?]
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146) [testng-7.3.0.jar:?]
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128) [testng-7.3.0.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_292]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
```
As for the second problem: https://app.travis-ci.com/github/apache/druid/jobs/574017552, that test got this error when Peon tried to boot up:
```bash
2022-06-19T21:53:38,414 ERROR [main] org.apache.druid.cli.CliPeon - Error when starting up.  Failing.
java.lang.reflect.InvocationTargetException: null

# due to

Caused by: java.nio.file.NoSuchFileException: /tmp/intermediary-segments/index_parallel_wikipedia_index_test_308123e8-c237-41eb-891f-0b6db6276c67 %Россия 한국 中国!?_omikabmi_2022-06-19T21:46:06.483Z
```
Full stacktrace:
```bash
2022-06-19T21:53:38,414 ERROR [main] org.apache.druid.cli.CliPeon - Error when starting up.  Failing.
java.lang.reflect.InvocationTargetException: null
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_332]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_332]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_332]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_332]
	at org.apache.druid.java.util.common.lifecycle.Lifecycle$AnnotationBasedHandler.start(Lifecycle.java:446) ~[druid-core-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	at org.apache.druid.java.util.common.lifecycle.Lifecycle.start(Lifecycle.java:341) ~[druid-core-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	at org.apache.druid.guice.LifecycleModule$2.start(LifecycleModule.java:152) ~[druid-core-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	at org.apache.druid.cli.GuiceRunnable.initLifecycle(GuiceRunnable.java:165) [druid-services-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	at org.apache.druid.cli.CliPeon.run(CliPeon.java:306) [druid-services-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	at org.apache.druid.cli.Main.main(Main.java:114) [druid-services-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
Caused by: java.io.UncheckedIOException: /tmp/intermediary-segments/index_parallel_wikipedia_index_test_308123e8-c237-41eb-891f-0b6db6276c67 %Россия 한국 中国!?_omikabmi_2022-06-19T21:46:06.483Z
	at org.apache.commons.io.FileUtils.listFiles(FileUtils.java:2153) ~[commons-io-2.11.0.jar:2.11.0]
	at org.apache.druid.indexing.worker.shuffle.LocalIntermediaryDataManager.lambda$discoverSupervisorTaskPartitions$3(LocalIntermediaryDataManager.java:200) ~[druid-indexing-service-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660) ~[?:1.8.0_332]
	at org.apache.druid.indexing.worker.shuffle.LocalIntermediaryDataManager.discoverSupervisorTaskPartitions(LocalIntermediaryDataManager.java:197) ~[druid-indexing-service-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	at org.apache.druid.indexing.worker.shuffle.LocalIntermediaryDataManager.start(LocalIntermediaryDataManager.java:134) ~[druid-indexing-service-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	... 10 more
Caused by: java.nio.file.NoSuchFileException: /tmp/intermediary-segments/index_parallel_wikipedia_index_test_308123e8-c237-41eb-891f-0b6db6276c67 %Россия 한국 中国!?_omikabmi_2022-06-19T21:46:06.483Z
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86) ~[?:1.8.0_332]
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102) ~[?:1.8.0_332]
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107) ~[?:1.8.0_332]
	at sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55) ~[?:1.8.0_332]
	at sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:144) ~[?:1.8.0_332]
	at sun.nio.fs.LinuxFileSystemProvider.readAttributes(LinuxFileSystemProvider.java:99) ~[?:1.8.0_332]
	at java.nio.file.Files.readAttributes(Files.java:1737) ~[?:1.8.0_332]
	at java.nio.file.FileTreeWalker.getAttributes(FileTreeWalker.java:225) ~[?:1.8.0_332]
	at java.nio.file.FileTreeWalker.visit(FileTreeWalker.java:276) ~[?:1.8.0_332]
	at java.nio.file.FileTreeWalker.walk(FileTreeWalker.java:322) ~[?:1.8.0_332]
	at java.nio.file.FileTreeIterator.<init>(FileTreeIterator.java:72) ~[?:1.8.0_332]
	at java.nio.file.Files.walk(Files.java:3574) ~[?:1.8.0_332]
	at org.apache.commons.io.file.PathUtils.walk(PathUtils.java:1044) ~[commons-io-2.11.0.jar:2.11.0]
	at org.apache.commons.io.FileUtils.streamFiles(FileUtils.java:2971) ~[commons-io-2.11.0.jar:2.11.0]
	at org.apache.commons.io.FileUtils.listFiles(FileUtils.java:2151) ~[commons-io-2.11.0.jar:2.11.0]
	at org.apache.druid.indexing.worker.shuffle.LocalIntermediaryDataManager.lambda$discoverSupervisorTaskPartitions$3(LocalIntermediaryDataManager.java:200) ~[druid-indexing-service-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660) ~[?:1.8.0_332]
	at org.apache.druid.indexing.worker.shuffle.LocalIntermediaryDataManager.discoverSupervisorTaskPartitions(LocalIntermediaryDataManager.java:197) ~[druid-indexing-service-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	at org.apache.druid.indexing.worker.shuffle.LocalIntermediaryDataManager.start(LocalIntermediaryDataManager.java:134) ~[druid-indexing-service-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	... 10 more
```

File: core/src/main/java/org/apache/druid/data/input/impl/CloudObjectInputSource.java
Comment (cryptoe) [inline core/src/main/java/org/apache/druid/data/input/impl/CloudObjectInputSource.java:132]: Please update the Javadoc for this method

File: docs/ingestion/native-batch-input-source.md
Comment (cryptoe) [inline docs/ingestion/native-batch-input-source.md:324]: IMHO this documentation can be improved. Do we have support for regex filtering? Wildcard seems just too limiting

File: core/src/main/java/org/apache/druid/data/input/impl/CloudObjectInputSource.java
Comment (cryptoe) [inline core/src/main/java/org/apache/druid/data/input/impl/CloudObjectInputSource.java:216]: Please add some validation to filter as its user input.

File: core/src/main/java/org/apache/druid/data/input/impl/CloudObjectInputSource.java
Comment (cryptoe) [inline core/src/main/java/org/apache/druid/data/input/impl/CloudObjectInputSource.java:133]: Can you please shed some light into why do we require this change for **objects** and **uris**?
These inputs are controlled by the user. Shouldn't the user not provide the files if they do not need to ingested. 

The filtering makes sense if we provide a **prefix** as then we do not know what can be present in the folder.

For that, I think we should explore: https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html#API_ListObjectsV2_ResponseSyntax and similar for other connectors. 
We can read up on prefix-based s3 lists for some part of the filtering at least. 
LMK if I am missing a use case that I do not understand.

File: core/src/main/java/org/apache/druid/data/input/impl/CloudObjectInputSource.java
Comment (abhishekagarwal87) [reply core/src/main/java/org/apache/druid/data/input/impl/CloudObjectInputSource.java:216]: what sort of validation do you expect @cryptoe? `filter` is nullable.

File: docs/ingestion/native-batch-input-source.md
Comment (abhishekagarwal87) [reply docs/ingestion/native-batch-input-source.md:324]:
```suggestion
|filter|A wildcard filter for files. See [here](http://commons.apache.org/proper/commons-io/apidocs/org/apache/commons/io/filefilter/WildcardFileFilter) for more information.|None|no|
```

File: core/src/main/java/org/apache/druid/data/input/impl/CloudObjectInputSource.java
Comment (cryptoe) [reply core/src/main/java/org/apache/druid/data/input/impl/CloudObjectInputSource.java:216]: I was thinking of a valid filter check like a valid regex, but seems like we do not have that helper method here: 
http://commons.apache.org/proper/commons-io/apidocs/org/apache/commons/io/filefilter/WildcardFileFilter

File: core/src/main/java/org/apache/druid/data/input/impl/CloudObjectInputSource.java
Comment (kfaraz) [inline core/src/main/java/org/apache/druid/data/input/impl/CloudObjectInputSource.java:59]: Nit: it maybe simpler to just call the other constructor from here.

**Review by kfaraz — CHANGES_REQUESTED** [review]
Thanks for adding this, @didip ! This seems very useful.
Overall, the changes look good to me.

As @abhishekagarwal87 and @cryptoe suggest, the docs probably need a bit more info. 

In addition to that, I also have a minor style concern. All the places that are now using an if-else on `StringUtils.isBlank(filter)` seem to have become somewhat less readable.

How about we always invoke the `FilenameUtils.wildcardMatch()` irrespective of whether the filter is blank or not? When filter is blank, just pass null, otherwise the actual value. The `FilenameUtils.wildcardMatch()` short-circuits to true when the filter is null, although the javadoc doesn't seem to say anything about it.

If this seems like an unnecessary dependency on the null-handling behaviour of an external library, we could just add a new method in some existing Druid utils (e.g. `FileUtils`) and use that instead. This method could internally call `FilenameUtils.wildcardMatch()` after doing a blank check.

**Comment by abhishekagarwal87** [comment]
FWIW, I think the docs don't need many changes. Pointing to the wildcard syntax is good enough. To your question @cryptoe about supporting regex filtering, this PR only adds support for wildcard filtering. the most common use case is to ignore meta files such as `_SUCCESS` / `.SUCCESS` that wild card filters can do.

**Comment by cryptoe** [comment]
> To your question @cryptoe about supporting regex filtering, this PR only adds support for wildcard filtering. the most common use case is to ignore meta files such as `_SUCCESS` / `.SUCCESS` that wild card filters can do.

According to the druid docs, 
<span>

property | description | default | required?
-- | -- | -- | --
type | This should be s3. | None | yes
uris | JSON array of URIs where S3 objects to be ingested are located. | None | uris or prefixes or objects must be set
prefixes | JSON array of URI prefixes for the locations of  S3 objects to be ingested. Empty objects starting with one of the given  prefixes will be skipped. | None | uris or prefixes or objects must be set
objects | JSON array of S3 Objects to be ingested. | None | uris or prefixes or objects must be set
properties | Properties Object for overriding the default S3 configuration. See below for more information. | None | No (defaults will be used if not given)

</span>

So we should only apply wildcard filtering when we specify **prefixes**. No ?
In the PR it seems like we are filtering for **uris** and **objects**. Hence the confusion.

**Comment by didip** [comment]
@cryptoe It is true that the filter is applied to all three. So this is my reasoning:

The `objects` is not just used by end users. It is also used after createSplit happened. So to ensure consistency, I applied the filtering on the objects as well.

From that perspective, I came to a conclusion that maybe it will be nice for the end-users if the behavior is consistent on all three. Imagine if all they are reading is just the docs page (and not the source code). The end-user would totally expect that `filter` will be applied on all `objects`, `prefixes`, and `uris`.  This is a win in terms of UX.

File: core/src/main/java/org/apache/druid/data/input/impl/CloudObjectInputSource.java
Comment (didip) [reply core/src/main/java/org/apache/druid/data/input/impl/CloudObjectInputSource.java:59]: @kfaraz I think that's too big of a change especially considering that `FilenameUtils.wildcardMatch` does not document the `null` behavior.

Also, I don't see why `FilenameUtils` cannot be used directly since other `commons` library are already used directly. Creating a method inside `FileUtils` would just add an unnecessary indirection.

Comment (kfaraz) [reply core/src/main/java/org/apache/druid/data/input/impl/CloudObjectInputSource.java:59]: Sure, @didip , sounds good.

**Review by kfaraz — APPROVED** [review]
LGTM

**Comment by abhishekagarwal87** [comment]
@didip - I think that makes sense. In fact, I was thinking along the very same lines even before you posted your comment :)

File: extensions-core/s3-extensions/src/main/java/org/apache/druid/data/input/s3/S3InputSource.java
Comment (cryptoe) [inline extensions-core/s3-extensions/src/main/java/org/apache/druid/data/input/s3/S3InputSource.java:221]: Sorry to play spoilsport but I see one potential issue with this PR. :(
IMHO we should push the filtering to getIterableObjectsFromPrefixes(). This way the filtered path will not be used in splitHintSpecCalculations. 
This logic may not work with people who have customer splitSpecs which do not take size into account. 

Also, I think we add the filtering to `getIterableObjectsFromPrefixes` in other input sources also like
1. AzureInputSource
2. OSSInputSource
3. GCSInputSource

we will not have the subtle bug where the path is used for calculating splits but is ultimately filtered out when returning to the user. 

@didip Appreciate your patience in answering the questions.

Comment (didip) [reply extensions-core/s3-extensions/src/main/java/org/apache/druid/data/input/s3/S3InputSource.java]: @cryptoe ok, I believe I have addressed your concern. In the latest codebase I moved the filtering logic into the respective `getIterableObjectsFromPrefixes()`. 

This cleaned up the code a lot, which I think address the readability problem that @kfaraz raised.

All of my new test cases still passed.

**Comment by didip** [comment]
@abhishekagarwal87 is travis stuck? I see that build is created but nothing is happening.

File: docs/ingestion/native-batch-input-source.md
Comment (cryptoe) [inline docs/ingestion/native-batch-input-source.md:243]:
```suggestion
|filter|A wildcard filter for files. See [here](http://commons.apache.org/proper/commons-io/apidocs/org/apache/commons/io/filefilter/WildcardFileFilter) for more information. Files matching the filter criteria are considered for ingestion. Files not matching the filter criteria are ignored.|None|no|
```

File: docs/ingestion/native-batch-input-source.md
Comment (cryptoe) [inline docs/ingestion/native-batch-input-source.md:324]:
```suggestion
|filter|A wildcard filter for files. See [here](http://commons.apache.org/proper/commons-io/apidocs/org/apache/commons/io/filefilter/WildcardFileFilter) for more information. Files matching the filter criteria are considered for ingestion. Files not matching the filter criteria are ignored.|None|no|
```

File: docs/ingestion/native-batch-input-source.md
Comment (cryptoe) [inline docs/ingestion/native-batch-input-source.md:147]:
```suggestion
|filter|A wildcard filter for files. See [here](http://commons.apache.org/proper/commons-io/apidocs/org/apache/commons/io/filefilter/WildcardFileFilter) for more information. Files matching the filter criteria are considered for ingestion. Files not matching the filter criteria are ignored.|None|no|
```

**Review by cryptoe — APPROVED** [review]
Changes LGTM. 
Thanks, @didip for the contribution.

File: docs/ingestion/native-batch-input-source.md
Comment (didip) [reply docs/ingestion/native-batch-input-source.md]: addressed

File: docs/ingestion/native-batch-input-source.md
Comment (didip) [reply docs/ingestion/native-batch-input-source.md]: addressed

File: docs/ingestion/native-batch-input-source.md
Comment (didip) [reply docs/ingestion/native-batch-input-source.md]: addressed

**Comment by didip** [comment]
@abhishekagarwal87 @cryptoe @kfaraz The TravisCI still failed on two of the unrelated issues I mentioned above. Can this PR be merged?

File: extensions-core/s3-extensions/src/main/java/org/apache/druid/data/input/s3/S3InputSource.java
Comment (abhishekagarwal87) [reply extensions-core/s3-extensions/src/main/java/org/apache/druid/data/input/s3/S3InputSource.java]: Good catch @cryptoe

**Comment by abhishekagarwal87** [comment]
Thank you @didip for your contribution.

**Pull request merged**
